test/framework/framework.go

```diff
@@ -0,0 +1,263 @@
+func (f *Framework) BeforeEach() {
+	f.buildClientsets()
+}
+
+func (f *Framework) ScaleSatefulSetDownToZero(statefulset *appsv1beta2.StatefulSet) error {
+	var zero int32 = 0
+	statefulset.Spec.Replicas = &zero
+	err := wait.Poll(PollInterval, PoolDeletionTimeout, func() (bool, error) {
+		// give it some time
+		_, err := f.KubeClient.AppsV1beta2().StatefulSets(statefulset.Namespace).Update(statefulset)
+		log.Infof("ScaleSatefulSetDownToZero.err: %v\n", err)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Now wait the number of replicas is really zero
+	return wait.Poll(PollInterval, PoolDeletionTimeout, func() (bool, error) {
+		// give it some time
+		result, err := f.KubeClient.AppsV1beta2().StatefulSets(statefulset.Namespace).Get(statefulset.Name, metav1.GetOptions{})
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+		fmt.Printf("result: %v\n", result.Status.CurrentReplicas)
+		if result.Status.CurrentReplicas == 0 {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+func (f *Framework) ScaleDeploymentDownToZero(deployment *appsv1beta2.Deployment) error {
+	var zero int32 = 0
+	deployment.Spec.Replicas = &zero
+	err := wait.Poll(PollInterval, PoolDeletionTimeout, func() (bool, error) {
+		// give it some time
+		_, err := f.KubeClient.AppsV1beta2().Deployments(deployment.Namespace).Update(deployment)
+		log.Infof("ScaleDeploymentDownToZero.err: %v\n", err)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Now wait the number of replicas is really zero
+	return wait.Poll(PollInterval, PoolDeletionTimeout, func() (bool, error) {
+		// give it some time
+		result, err := f.KubeClient.AppsV1beta2().Deployments(deployment.Namespace).Get(deployment.Name, metav1.GetOptions{})
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+		fmt.Printf("result: %v\n", result.Status.AvailableReplicas)
+		if result.Status.AvailableReplicas == 0 {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+func WaitUntilDeleted(delFnc func() error, getFnc func() error) error {
+	return wait.Poll(PollInterval, PoolDeletionTimeout, func() (bool, error) {
+
+		err := delFnc()
+		log.Infof("del.err: %v\n", err)
+		if err != nil {
+			if strings.Contains(err.Error(), "object is being deleted") {
+				return false, nil
+			}
+			if strings.Contains(err.Error(), "not found") {
+				return true, nil
+			}
+			return false, nil
+		}
+
+		err = getFnc()
+		log.Infof("get.err: %v\n", err)
+		if err != nil && strings.Contains(err.Error(), "not found") {
```

Comment:
Reduce as before. I'll stop commenting on this - please take a look through the rest and reduce where possible.